### PR TITLE
Add validateSchedulesFile method

### PR DESCRIPTION
**Pull request by faiyaz26**
Also updating the validation error message to make it more understandable to user.

### CODE REVIEW

File: api/commands/validate.ts [hunk 1]
```diff
@@ -1,16 +1,51 @@
...
 function validateEmail(email: string) {
   return EMAIL_REGEX.test(email);
 }
 
-export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, compiledGraph: dataform.ICompiledGraph) {
+export function validateSchedulesFile(
+  compiledGraph: dataform.ICompiledGraph,
+  options: IOptions
+): string[] {
+  if (!options || !options.projectDir) {
```
Comment (BenBirt) [inline api/commands/validate.ts]: this seems odd to me. the user shouldn't ever be calling `validateSchedulesFile` without a`projectDir`. it seems like invalid usage of the API to me, so let's not support it?

File: api/commands/validate.ts [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+    return [];
+  }
+  const filePath = path.resolve(path.join(options.projectDir, SCHEDULES_JSON_PATH));
+  if (!fs.existsSync(filePath)) {
```
Comment (BenBirt) [inline api/commands/validate.ts]: this is OK for me, but I think we should rename the function to `validateSchedulesFileIfExists` - as a caller i'd expect this function to fail if it couldn't read the relevant file.

File: api/commands/validate.ts [hunk 3]
```diff
@@ -1,16 +1,51 @@
...
+  }
+  const content = fs.readFileSync(filePath, "utf8");
+  try {
+    const scheduleJsonObj = JSON.parse(content);
+    return validateSchedules(
+      dataform.schedules.SchedulesJSON.create(scheduleJsonObj),
+      compiledGraph
+    );
+  } catch (err) {
+    return [`${SCHEDULES_JSON_PATH} does not contain valid json.`];
```
Comment (BenBirt) [inline api/commands/validate.ts]: `... does not contain valid JSON conforming to the SchedulesJSON schema.`

File: api/commands/validate.ts [hunk 4]
```diff
@@ -31,37 +66,52 @@ export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, c
     if (!schedule.name) {
       errors.push("Schedule name is required.");
     }
+    if (schedule.name && schedule.name.trim().length === 0) {
+      errors.push("Schedule name can not be blank.");
```
Comment (BenBirt) [inline api/commands/validate.ts]: `Schedule name must not be empty.`

File: api/commands/validate.ts [hunk 5]
```diff
@@ -31,37 +66,52 @@ export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, c
...
+    }
     if (uniqueNames.has(schedule.name)) {
-      errors.push(`"${schedule.name}" is not unique. All schedule names must be unique.`);
+      errors.push(
+        `Schedule name: "${schedule.name}" is not unique. All schedule names must be unique.`
```
Comment (BenBirt) [inline api/commands/validate.ts]: nit: remove the `:`

File: api/commands/validate.ts [hunk 6]
```diff
@@ -31,37 +66,52 @@ export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, c
...
-      errors.push("Cron expression is required.");
+      errors.push(`Cron expression is required on ${schedule.name}.`);
     }
 
     try {
       const _ = cronParser.parseExpression(schedule.cron);
     } catch (e) {
-      errors.push(`"${schedule.cron}" is not a valid cron expression.`);
+      errors.push(
+        `Cron expression: "${schedule.cron}" is not a valid on schedule: ${schedule.name}.`
```
Comment (BenBirt) [inline api/commands/validate.ts]: `Schedule "${schedule.name}" contains an invalid cron expression "${schedule.cron}".`

File: api/commands/validate.ts [hunk 7]
```diff
@@ -31,37 +66,52 @@ export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, c
...
+      );
     }
 
     if (schedule.options && schedule.options.actions) {
       schedule.options.actions.forEach(action => {
         if (allActionNames.indexOf(action) < 0) {
-          errors.push(`"${action}" doesn't exist in the project.`);
+          errors.push(
+            `Action: ${action}" included on schedule: ${
```
Comment (BenBirt) [inline api/commands/validate.ts]: nit: remove both of the `:`, and s/on schedule/in schedule/

File: api/commands/validate.ts [hunk 8]
```diff
@@ -31,37 +66,52 @@ export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, c
...
-      })
+      });
     }
 
     if (schedule.notification && schedule.notification.emails) {
       schedule.notification.emails.forEach(email => {
         if (!validateEmail(email)) {
-          errors.push(`"${email}" is not a valid email address.`);
+          errors.push(
+            `Email address:"${email}" included on schedule: ${
```
Comment (BenBirt) [inline api/commands/validate.ts]: `Schedule "${schedule.name}" contains an invalid email address "${email}".`

File: tests/utils/BUILD [hunk 9]
```diff
@@ -5,4 +5,9 @@ load("@npm_bazel_typescript//:index.bzl", "ts_library")
 ts_library(
     name = "utils",
     srcs = glob(["**/*.ts"]),
+    deps = [
+        "@npm//@types/mocha",
```
Comment (BenBirt) [inline tests/utils/BUILD]: this isn't needed

File: api/commands/validate.ts (hunk 1 above)
Comment (faiyaz26) [reply api/commands/validate.ts]: I got the influence from here: https://github.com/dataform-co/dataform/blob/master/api/commands/query.ts#L41

File: tests/utils/BUILD (hunk 9 above)
Comment (faiyaz26) [reply tests/utils/BUILD]: If I remove this, I get this error:
`tests/utils/fixtures.ts:12:25 - error TS2503: Cannot find namespace 'Mocha'.`

Comment (BenBirt) [reply tests/utils/BUILD]: ah, my bad, of course that's needed. i looked at the file's imports and didn't see anything, but yes 'Mocha' needs to be defined. fascinating, they must be putting it on the global object or something!

**Pull request merged**